src/bus_monitor.cpp

```diff
@@ -461,60 +461,59 @@ void BootProgressCode::progressCodeCallBack(sdbusplus::message_t& msg)
                 return;
             }
 
-            // To detect if there is a need to save SRCs and hexwords in func 11
-            // to 13, check for array data filled with space.
-            if (hexWordArray.at(0) != 0x20)
+            // store the primary SRC.
+            std::string hexWordsWithSRC =
+                std::string(byteArray.begin(), byteArray.end());
+
+            /** Byte 4 in the Base SRC structure gives the total number of hex
+             *words. As per SPEC, word 1 is used as part of the header. The
+             *extended hex SRC words which we are interested in starts from
+             *word 2. To calculate the count of valid extended hex SRC words,
+             *remove the count for first word (which is a part of the header).
+             **/
+            types::Byte validHexWords = hexWordArray.at(3) - 1;
```

Comment:
why is valid hex words checked. 
I had mentioned not to use it

Reply:
removed it